src/javascript.rs

```diff
@@ -111,6 +118,40 @@ fn module_function<'a>(
     ])
 }
 
+fn external_function<'a, T>(
+    public: bool,
+    name: &'a str,
+    arguments: &'a [ExternalFnArg<T>],
+    module: &'a str,
+    fun: &'a str,
+) -> Output<'a> {
+    let head = if public {
+        "export function "
+    } else {
+        "function "
+    };
+    let arguments = wrap_args(arguments.iter().enumerate().map(|a| match a {
+        (index, ExternalFnArg { label, .. }) => {
+            let arg_name = label.clone().unwrap_or(format!("arg{}", index));
+            Document::String(arg_name)
+        }
```

Comment:
I think we can remove the cloning of the name here

```suggestion
        (index, ExternalFnArg { label, .. }) => {
            label.as_ref().map(Documentable::to_doc).unwrap_or_else(|| Document::String(format!("arg{}", index)));
        }
```

Reply:
Have added and got.

```
the trait `pretty::Documentable<'_>` is not implemented for `&std::string::String`
```

Reply:
OK, will need to turn the string into a str first
```rust
label.as_ref().map(|l| l.as_str().to_doc()).unwrap_or_else(|| Document::String(format!("arg{}", index)));
```